### ISSUE TITLE
Have an unconnected frame

### ISSUE THREAD

**Issue by nstarman** [astropy/astropy#13418]
### Description

Simulations are often run without a particular frame in mind, it's just "out there". The astropy coordinates data structures are still useful, but can't be used because we don't have a "null" frame. I'd like a Null/Un/SimulationFrame which is not connected to the frame transform graph.

**Comment by pllim** [comment]
Then why not just use *a* frame instead of a special one?

**Comment by nstarman** [comment]
As in subclass BaseCoordinateFrame and define a new frame object? Or as in just use, e.g. Galactocentric?

**Comment by pllim** [comment]
Doesn't matter right since everything is fake to begin with? 😸

**Comment by ayshih** [comment]
Yeah, I'd just define a subclass of `BaseCoordinateFrame`.  In fact, I'll point out that `BaseEclipticFrame` and `BaseRADecFrame` are two examples in Astropy of transform-free frames.

**Comment by nstarman** [comment]
In my own code I have a frame called ``UnFrame`` which I use when no particular should be used. It's just a subclass of `BaseCoordinateFrame` that I made a singleton. I'm not sure if the singleton is what we want, but having an official null frame seems useful. E.g. https://astropy-healpix.readthedocs.io/en/latest/api/astropy_healpix.HEALPix.html#astropy_healpix.HEALPix has to go through some contortions to support if a user isn't in a particular coordinate frame. Instead they could use the null frame.
And not having to roll one's own null frame for simulations is nice.
Besides subclassing `BaseCoordinateFrame ` I think we just modify the ``transform_to`` method to raise an Exception.
